PGM/src/main/java/tc/oc/pgm/shop/SimplePurchaseTracker.java

```diff
@@ -0,0 +1,141 @@
+        startGlobalStrategy(purchasable, player, purchasable.getPurchaseFilter());
+    }
+
+    @Override
+    public void startIndividualStrategy(MatchPlayer player, Purchasable purchasable, Filter filter) {
+        PlayerPaymentStrategy strategy = new PlayerPaymentStrategy(purchasable, filter, player, crossParty, persistent);
+        individualStrategies.put(player, strategy);
+        contribute(strategy, player);
+    }
+
+    @Override
+    public void startPartyStrategy(Party party, MatchPlayer player, Purchasable purchasable, Filter filter) {
+        PartyPaymentStrategy strategy = new PartyPaymentStrategy(purchasable, filter, party);
+        partyStrategies.put(party, strategy);
+        contribute(strategy, player);
+    }
+
+    @Override
+    public void startGlobalStrategy(Purchasable purchasable, MatchPlayer player, Filter filter) {
+        GlobalPaymentStrategy strategy = new GlobalPaymentStrategy(purchasable, filter);
+        globalStrategies.add(strategy);
+        contribute(strategy, player);
+    }
+
+    @Override
+    public Optional<PaymentStrategy> getOngoingStrategy(MatchPlayer player, Purchasable purchasable) {
+        return individualStrategies.values().stream().filter(e ->
+            e.getOwner().equals(player) && e.getPurchasable().equals(purchasable)
+        ).findFirst();
+    }
+
+    @Override
+    public Optional<PaymentStrategy> getOngoingStrategy(Party party, Purchasable purchasable) {
+        if (!(party instanceof Party))
+            return Optional.empty();
+        return partyStrategies.values().stream().filter(e ->
+                e.getOwner().equals(party) && e.getPurchasable().equals(purchasable)
+        ).findFirst();
+    }
+
+    @Override
+    public Optional<PaymentStrategy> getOngoingStrategy(Purchasable purchasable) {
+        return partyStrategies.values().stream().filter(e ->
+                e.getPurchasable().equals(purchasable)
+        ).findFirst();
+    }
+
+    @Override
+    public void contribute(PaymentStrategy strategy, MatchPlayer player) {
+        strategy.contribute(player);
+        if (strategy.isComplete()) {
+            switch (strategy.getPurchasable().getType()) {
+                case GLOBAL:
+                    globalStrategies.remove(strategy);
+                    strategy.getPurchasable().reward(player.getMatch().getPlayers());
+                    break;
+                case INDIVIDUAL:
+                    individualStrategies.remove(player, strategy);
+                    strategy.getPurchasable().reward(Sets.newHashSet(player));
+                    break;
+                case PARTY:
+                    partyStrategies.remove(strategy.getOwner(), strategy);
+                    strategy.getPurchasable().reward(((Party)strategy.getOwner()).getPlayers());
+                    break;
+            }
+        }
+    }
+
+    @EventHandler
+    public void onChange(PlayerChangePartyEvent event) {
```

Comment:
I'm guessing there are issues if I switch from one competitor to another. You will need to test that and sort the bugs.